check_test.go

```diff
@@ -401,4 +401,19 @@ func TestCheck_NoFlapping(t *testing.T) {
 	assert.Equal(t, 0, originalCheck.failureCounter)
 	assert.Equal(t, 0, originalCheck.successCounter)
 	assert.Equal(t, api.HealthCritical, originalCheck.Status)
+
+	runner.UpdateCheck(id, api.HealthPassing, "")
```

Comment:
Thanks for adding a test to check resetting counter! Would you mind adding a comment above to share what you are testing and help split apart this set of code (line 38 for an example)? It will help future people who look at this code (myself included!).

Reply:
I added a comment to the code.